src/borg/archiver.py

```diff
@@ -744,8 +744,17 @@ def do_extract(self, args, repository, manifest, key, archive):
         hardlink_masters = {} if partial_extract else None
```

Comment:
This needs to be something like:

```
         hardlink_masters = {} if partial_extract or not has_link else None 
```

Reply:
I'll fix this, thanks!

Reply:
fixed.

do you think it is ready to merge now?

Reply:
update: reread the ticket, answer seems "yes".